doc/rst/source/debug.rst

```diff
@@ -119,3 +121,60 @@ Applications).  Xcode may change as versions change; the images below is for Xco
    .. figure:: /_images/xcode-8.*
       :width: 100%
       :align: center
+
+Debug PyGMT in Xcode on macOS
+------------------------------
+
+**Note**: Tested with Xcode 12.0.  Install PyGMT following the official instructions (if you already have the dev version then you may just
+need to cd into your pygmt dir and call git pull):
+
+#. Add conda-forge channel::
+
+    conda config --prepend channels conda-forge
+    **Note**: The next step is different from the PyGMT official instructions, because we want to use the GMT dev version
+    conda create --name pygmt python=3.8 pip numpy pandas xarray netcdf4 packaging
+
+#. Activate the PyGMT environment::
+
+    conda activate pygmt
+
+#. Install Pygmt in editable/development mode::
+
+    cd pygmt
+    make install
```

Comment:
@weiji14 Do you think if it's easier to simply run the following command to install pygmt and its dependencies, instead of creating a virtual environment using conda: 
```
cd pygmt
pip install -e .
```

Reply:
Yes it would be easier, but this would contaminate the system python. I.e. ok for the occasional Python user like Paul, but not for those of us who might work with different virtual environments. But we can put `conda activate pygmt` as **optional** with this caveat in mind.

I would also go with `pip install --editable .` instead of `pip install -e .` just for readability, again for casual Python users :wink:

Reply:
Could you of you make the changes to the debug.rst file to cover these cases?  Then we can call it a wrap.